src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -182,64 +203,46 @@ int MPL_gavl_tree_create(void (*free_fn) (void *), MPL_gavl_tree_t * gavl_tree)
     goto fn_exit;
 }
 
-int MPL_gavl_tree_insert(MPL_gavl_tree_t gavl_tree, const void *addr, uintptr_t len,
-                         const void *val)
+static void gavl_tree_search_node(gavl_tree_s * gavl_tree_iptr, uintptr_t addr, uintptr_t len,
+                                  int mode, gavl_tree_node_s ** ret_node)
```

Comment:
I prefer to return value through function input parameters; usually, the value returned through function is error code. Although the current function does not return anything, in future, we may add an error code return.
I didn't name it `gavl_tree_search` because I want to avoid mixing naming with `MPL_gavl_tree_search`, we want to point out it is an internal function and only search a node.

Reply:
For "static" internal function, the convention is looser and we should try to make the code simpler and more readable. As long as it is easy to read, whenever you want to expand or change the functionality, you can change it, and the impact is local anyway.